Riot/Modules/Home/AllChats/AllChatsViewController.swift

```diff
@@ -87,41 +104,230 @@ class AllChatsViewController: HomeViewController {
-                     discoverabilityTitle: VectorL10n.roomRecentsCreateEmptyRoom,
-                     handler: { [weak self] action in
-                self?.createNewRoom()
-            }),
-            UIAction(title: VectorL10n.roomRecentsStartChatWith,
-                     image: Asset.Images.sideMenuActionIconFeedback.image,
-                     discoverabilityTitle: VectorL10n.roomRecentsStartChatWith,
-                     handler: { [weak self] action in
-                self?.startChat()
-            })
-        ])
-        
-        toolbar.items = [
-            UIBarButtonItem(image: UIImage(systemName: "ellipsis.circle"), menu: AllChatsActionProvider().menu),
+        self.tabBarController?.navigationItem.rightBarButtonItem = UIBarButtonItem(image: UIImage(systemName: "line.3.horizontal.decrease.circle"), menu: AllChatsActionProvider().menu)
+    }
+
+    private func updateUI() {
+        let currentSpace = self.dataSource?.currentSpace
+        self.tabBarController?.title = currentSpace?.summary?.displayname ?? VectorL10n.allChatsTitle
+        
+        updateToolbar(with: editActionProvider.updateMenu(with: mainSession, parentSpace: currentSpace, completion: { [weak self] menu in
+            self?.updateToolbar(with: menu)
+        }))
+    }
+    
+    private func updateToolbar(with menu: UIMenu) {
+        let currentSpace = self.dataSource?.currentSpace
+        self.navigationController?.isToolbarHidden = false
+        self.tabBarController?.setToolbarItems([
+            UIBarButtonItem(image: Asset.Images.homeMySpacesAction.image, style: .done, target: self, action: #selector(self.showSpaceSelectorAction(sender: ))),
             UIBarButtonItem.flexibleSpace(),
-            UIBarButtonItem(image: UIImage(systemName: "square.and.pencil"), menu: editMenu)
-        ]
+            UIBarButtonItem(image: UIImage(systemName: currentSpace == nil ? "square.and.pencil" : "ellipsis.circle"), menu: menu)
+        ], animated: true)
+    }
+    
+    private func showCreateSpace(parentSpaceId: String?) {
+        let coordinator = SpaceCreationCoordinator(parameters: SpaceCreationCoordinatorParameters(session: self.mainSession, parentSpaceId: parentSpaceId))
+        let presentable = coordinator.toPresentable()
+        self.present(presentable, animated: true, completion: nil)
+        coordinator.callback = { [weak self] result in
+            guard let self = self else {
+                return
+            }
+            
+            self.createSpaceCoordinator?.toPresentable().dismiss(animated: true) {
+                self.createSpaceCoordinator = nil
+                switch result {
+                case .cancel:
+                    break
+                case .done(let spaceId):
+                    self.switchSpace(withId: spaceId)
+                }
+            }
+        }
+        coordinator.start()
+        
+        self.createSpaceCoordinator = coordinator
```

Comment:
This class seems to have 3 ways it handles coordinators.
- Here the coordinator has a property and isn't added to `childCoordinators`
- `showSpaceInvite`/`Members`/`Settings`/`Leave` all add their coordinators to `childCoordinators`
- `showSpaceSelectorAction` uses a bridge presenter to manage the coordinator.

I think you may have mentioned about adding an `AllChatsCoordinator` in the future which would solve this so just wanted to check.

Reply:
I mentioned the future `AllChatsCoordinator` as child coordinators shouldn't be managed from the `AllChatsViewController` but from its parent coordinator.

Nevertheless, I've changed this method so it now use the `childCoordinators` for consistency purpose :)